test/snoowrap.spec.js

```diff
@@ -219,6 +220,45 @@ describe('snoowrap', function () {
     });
   });
 
+  describe('.fromApplicationOnlyAuth', () => {
+    it('throws a TypeError if no userAgent is provided in node', function () {
+      if (isBrowser) {
+        return this.skip();
+      }
+      expect(() => {
+        snoowrap.fromApplicationOnlyAuth({clientId: 'bar', deviceId: 'baz'});
+      }).to.throw(TypeError);
+    });
+    it('throws a TypeError if no clientId is provided', () => {
+      expect(() => {
+        snoowrap.fromApplicationOnlyAuth({userAgent: 'bar'});
+      }).to.throw(TypeError);
+    });
+    it('throws a TypeError if no grantType is provided', () => {
+      expect(() => {
+        snoowrap.fromApplicationOnlyAuth({userAgent: 'bar'});
+      }).to.throw(TypeError);
+    });
+    it('returns a Promise for a valid installed requester', async () => {
+      const newRequester = await snoowrap.fromApplicationOnlyAuth({
+        userAgent: oauthInfo.user_agent,
+        grantType: snoowrap.grantType.INSTALLED_CLIENT,
+        clientId: oauthInfo.installed_app_client_id,
+        deviceId: oauthInfo.device_id
+      });
+      expect(await newRequester.getHot('redditdev', {limit: 1})).to.have.lengthOf(1);
+    });
+    it('returns a Promise for a valid client requester', async () => {
```

Comment:
Could "it returns a promise" can be changed to "it returns a snoowrap instance"? Seems more logical IMO

Reply:
Done, but on second thought, is this true?

Doesn't this return a listing, ie, not a snoowrap instance?

`await newRequester.getHot('redditdev', {limit: 1})`

Also last test in `.fromAuthCode` also mentions returning a Promise, so whatever the final decision is, we should probably make it consistent.

Reply:
Well, I would at first glance of returning a promise imagine that it tests whether we get a promise or not for something, not that we test that we get a valid snoowrap instance that can execute queries.

Checking whether we can query the API in the test is just a way to make sure it works. We don't care whether or not we get a Listing as long as we can be certain the snoowrap instance works. 

I'd change the `.fromAuthCode` as well to be clearer. Feel free to do so now or I'll do it after merge :)